src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/DynamicHelpers.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.VisualStudio.Debugger.Metadata;
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    internal static class DynamicHelpers
+    {
+        private static readonly bool[] TrueArray = new[] { true };
+
+        public static bool[] GetDynamicFlags(this IList<CustomAttributeData> attributes)
+        {
+            foreach (var attribute in attributes)
+            {
+                if (attribute.Constructor.DeclaringType.IsType("System.Runtime.CompilerServices", "DynamicAttribute"))
+                {
+                    var arguments = attribute.ConstructorArguments;
+                    if (arguments.Count == 0)
+                    {
+                        return TrueArray;
+                    }
+                    else if (arguments.Count == 1)
+                    {
+                        var argumentType = arguments[0].ArgumentType;
+                        if (argumentType.IsArray && argumentType.GetElementType().IsBoolean())
+                        {
+                            // Per https://msdn.microsoft.com/en-us/library/system.reflection.customattributetypedargument.argumenttype(v=vs.110).aspx,
+                            // if ArgumentType indicates an array, then Value will actually be a ReadOnlyCollection.
+                            var collection = (ReadOnlyCollection<CustomAttributeTypedArgument>)arguments[0].Value;
+                            var builder = ArrayBuilder<bool>.GetInstance(collection.Count);
+                            foreach (var arg in collection)
+                            {
+                                builder.Add((bool)arg.Value);
+                            }
+                            return builder.ToArrayAndFree();
+                        }
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        public static T[] Skip<T>(this T[] oldArray, int count)
+        {
+            var oldLength = oldArray.Length;
+
+            if (count > oldLength)
+            {
+                throw new System.ArgumentOutOfRangeException(nameof(count), count, $"Cannot skip more than {oldLength} elements.");
+            }
+
+            if (count < 0)
+            {
+                throw new System.ArgumentOutOfRangeException(nameof(count), count, $"Cannot skip a negative number of elements.");
+            }
+
+            if (count == oldLength)
+            {
+                return new T[0];
+            }
```

Comment:
I suspect I pulled this out since I was expecting there to be a helper for reusing empty array instances.  I can't seem to find a portable one, so I'll delete this case.